android-components/components/feature/readerview/src/main/assets/extensions/readerview/readerview.js

```diff
@@ -179,6 +179,10 @@ class ReaderView {
           readingTimeString = `${readingTimeMinsFast} - ${readingTimeString}`;
         }
         return readingTimeString;
+      } else {
+        // This block handles the case when `parts` array does not have length 3,
+        // For some languages parts may not have length 3.
+        return parts.map(e => e.value).join(" ");
```

Comment:
I've reviewed this in the original PR here: https://github.com/mozilla-mobile/android-components/pull/11951#issuecomment-1282791718

The main problem is that we do not have reading time estimates for this language, and therefore even the parts you do get back here are incorrect. This is because we fall back to estimating the reading time in English, and incorrectly try to present it as reading time in the article's language.

So, you're right that there are languages for which we won't have all three parts, but there's a bigger problem, because the reading time estimates for these languages are also not correct.

On Firefox Desktop, we also fall back to reading time estimates in English, but then also display it in English if the language is not supported. So, I recommend we do the same here, see the fix in my [comment](https://github.com/mozilla-mobile/android-components/pull/11951#issuecomment-1282791718) above.

Reply:
Alternatively, instead of displaying an incorrect estimate, maybe we shouldn't display one at all?

Reply:
Thank you for the review @csadilek, I will update the PR according to your comment which will also make the app consistent with the desktop behavior.

Reply:
OK, thanks, I think we can land this next week. Can you rebase and squash your commits into one?

Reply:
Thank you for the review, I will make the needful changes as soon as I can, thanks.